packages/core/src/Machine.ts

```diff
@@ -50,12 +55,12 @@ export function Machine<
 }
 
 export function createMachine<
-  TModel extends Model<any, any, any>,
+  TModel extends Model<any, any, any, any>,
   TContext = ModelContextFrom<TModel>,
   TEvent extends EventObject = ModelEventsFrom<TModel>,
   TTypestate extends Typestate<TContext> = { value: any; context: TContext }
 >(
-  config: MachineConfig<TContext, any, TEvent>,
+  config: MachineConfig<TContext, any, TEvent, ModelActionsFrom<TModel>>,
```

Comment:
I would move this to the list of generics - so all the "computed" stuff from the model would live there